libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2034,6 +2036,21 @@ void TypeChecker::endVisit(Literal const& _literal)
 				"For more information please see https://solidity.readthedocs.io/en/develop/types.html#address-literals"
 			);
 	}
+	if (_literal.isHexNumber() && _literal.subDenomination() != Literal::SubDenomination::None)
+	{
+		if (v050)
+			m_errorReporter.fatalTypeError(
+				_literal.location(),
+				"Hexadecimal numbers cannot be used with unit denominations. "
+				"You can use an expression of the form '0x1234 * 1 day' instead."
```

Comment:
Just as a note, usually we've used double quotes for all these, i.e. `form \"0x1234 * 1 day\" instead`

Reply:
@ekpyron can you please make this small change?